hnn_core/network.py

```diff
@@ -849,6 +860,11 @@ def _reset_drives(self):
         for drive_name in self.external_drives.keys():
             self.external_drives[drive_name]['events'] = list()
 
+    def _reset_rec_array(self):
```

Comment:
I have thought about these `reset` methods. I think this is an indication that these objects don't belong as attributes of the `Network` object. They should be instantiated in `simulate_dipole` and returned separately